SharedLibs/TileMapHelper/src/model/TileMapModel.ts

```diff
@@ -352,25 +504,243 @@ export class EditableTileMapLayer extends AbstractEditableLayer {
+      );
+    }
+    if (columnsToPop > 0 || columnsToShift > 0) {
+      this._tiles.forEach((row, rowIndex) => {
+        this._tiles[rowIndex] = this._tiles[rowIndex].slice(
+          columnsToShift,
+          columnsToPop ? -columnsToPop : undefined
+        );
+      });
+    }
+    // TODO: Instead of setting the dimensions directly, should it call a method on
+    // EditableTileMap that will iterates over all the layers to change their dimensions?
+    this.tileMap.setDimensionX(
+      initialColumnCount - columnsToPop - columnsToShift
+    );
+    this.tileMap.setDimensionY(initialRowCount - rowsToPop - rowsToShift);
+  }
+
+  increaseDimensions(
+    columnsToAppend: number,
+    columnsToUnshift: number,
+    rowsToAppend: number,
+    rowsToUnshift: number
+  ) {
+    const initialRowCount = this._tiles.length;
+    const initialColumnCount = this._tiles[0].length;
+    if (columnsToAppend > 0 || columnsToUnshift > 0) {
+      this._tiles.forEach((row, rowIndex) => {
+        const newRow = new Int32Array(
+          initialColumnCount + columnsToAppend + columnsToUnshift
+        ).fill(0);
+        newRow.set(row, columnsToUnshift);
+        this._tiles[rowIndex] = newRow;
+      });
+    }
+    if (rowsToAppend > 0 || rowsToUnshift > 0) {
+      this._tiles.unshift(
+        ...new Array(rowsToUnshift)
+          .fill(0)
+          .map(() =>
+            new Int32Array(
+              initialColumnCount + columnsToAppend + columnsToUnshift
+            ).fill(0)
+          )
+      );
+
+      this._tiles.length = initialRowCount + rowsToAppend + rowsToUnshift;
+
+      for (
+        let rowIndex = initialRowCount + rowsToUnshift;
+        rowIndex < this._tiles.length;
+        rowIndex++
+      ) {
+        this._tiles[rowIndex] = new Int32Array(
+          initialColumnCount + columnsToAppend + columnsToUnshift
+        ).fill(0);
+      }
+    }
+    // TODO: Instead of setting the dimensions directly, should it call a method on
+    // EditableTileMap that will iterates over all the layers to change their dimensions?
+    this.tileMap.setDimensionX(
+      initialColumnCount + columnsToAppend + columnsToUnshift
+    );
+    this.tileMap.setDimensionY(initialRowCount + rowsToAppend + rowsToUnshift);
+  }
+
   /**
    * @param x The layer column.
    * @param y The layer row.
    * @param tileId The tile.
+   * @param options Flipping options.
    */
-  setTile(x: integer, y: integer, tileId: integer): void {
+  setTile(
+    x: integer,
+    y: integer,
+    tileId: integer,
+    options:
+      | {
+          flipVertically: boolean;
+          flipHorizontally: boolean;
+          flipDiagonally: boolean;
+        }
+      | undefined = {
+      flipVertically: false,
+      flipHorizontally: false,
+      flipDiagonally: false,
+    }
+  ):
+    | {
+        unshiftedRows: number;
+        unshiftedColumns: number;
+        appendedRows: number;
+        appendedColumns: number;
+      }
+    | undefined {
```

Comment:
The runtime don't use parameters as object for efficiency reasons.

Reply:
We can make exceptions if it's not intensively used in a frame. If too many arguments, it might be worth consolidating them in an option object (that could even be cached if the runtime thinks it's better).

Reply:
So an array? Or does the array have the same issue?

Reply:
The method does not use objects as parameters but still returns an object.

Reply:
Existing code was using `FlippingHelper` outside to avoid this issue. It seems you go back to this.
About the returned value, I think that either:
- the method should expand in "negative" to keep the same indexes and it wouldn't need to give the info back
- the method should not handle expansion at all and the caller should do it itself and thus doesn't need to get this info back